chore: incremental 7.0 merge back into edge

shlokamin opened the pull request:
# Overview
Incremental merge of 7.0 back into edge to keep things up to data

I made a branch off of `chore_release-7.0.0` so i don't have to keep playing whack a mole and asking engineers to not merge commits (thx for the idea @SyntaxColoring!)

SyntaxColoring committed:
feat(api): Alias `del deck[...]` to `moveLabware(..., OFF_DECK)` (#13239)

b-cooper committed:
fix(app): fix temperature module wait for temp text (#12738)

The wait for target temperature command for the temperature module improperly interpolates a
temperature value, despite the fact that only the target temperature previously set will be awaited

Closes RQA-749

b-cooper committed:
fix(app-shell-odd, discovery-client): add option for MDNS in discovery client for use in odd (#13363)

Because the ODD app shell is only interacting with the local robot, allow it to configure it's
discovery client instance to ignore the MDNS poll.

jerader committed:
fix(app): labware setup renders and location fixes (#13371)

closes RQA-1385 closes RQA-1383

ncdiehl11 committed:
fix(app): display trash name on droptip command text (#13337)

* fix(app): display Fixed Trash for dropped tips on Flex

* fix(app): expand labware display name checks for Fixed Trash

* fix: remove req for gripper+pipette data interfaces calibratedoffsets

* fix(app): allow instrument calibratedOffset to be undefined

mjhuff committed:
fix(app): Fix white screen on RobotDashboard when clicking protocol (#13367)

* fix(app): fix white screen issue on RobotDashboard

Adds loading state when pressing a recently ran protocol. Fixes routing error.

koji committed:
fix(app): Fix RunFailed modal WhiteScreen and plus alpha (#13312)

* fix(app): fix run failed modal for desktop white screen issue and add utc label to timestamp

ncdiehl11 committed:
fix(app): fix playbutton click behavior if not ready (#13352)

* fix: remove req for gripper+pipette data interfaces calibratedoffsets

* fix(app): fix behavior of clicking the ODD protocol setup deactivated play button

* fix(app/): fix linting error for possibly null value

* fix(app): fix(app/): fix linting error for possibly null value

* merge in new pipette liquid defaults

* fix(app): check whether the play button is ready before confirming attachment

* fix(app): revert to checking attached modules method (can be null)

* fix(app): remove check for heater shaker shaking

* fix(app): remove unused imports

smb2268 committed:
fix(app): ensure current run id is one created before issuing commands (#13373)

fix RLIQ-1377

sanni-t committed:
fix(api): fix run cancellation race condition bug (#13339)

* remove call to hardware halt from engine.stop
* make halt a part of engine finish
* make sure execution manager cancellation completes before reset is called

koji committed:
fix(app): fix view network details data fetch issue (#13293)

* fix(app): fix view network details data fetch issue

jbleon95 committed:
fix(api): pick_up_tip in 2.14+ protocols does not pick up tips before starting tip (#13156)

fix regression in starting tip pick up behavior

ryanthecoder committed:
feat(shared-data): Increase p1000 single dispense flow rates (#13382) (#13385)

Co-authored-by: Andy Sigler <andrewsigler1@gmail.com>

jbleon95 committed:
feat(shared-data, app): separate definition for 96-channel tip rack adapter (#13379)

split definition for 96 channel tip rack adapter and logic to support that for LPC

---------

Co-authored-by: Jethary <jethary@opentrons.com>
Co-authored-by: Brian Cooper <brian@opentrons.com>

sanni-t committed:
fix(engine): fix calculation for labware height on modules (#13364)

koji committed:
feat(app): add module calibrate button to protocol setup (#13380)

* feat(app): add calibrate button to Desktop app protocol setup

mjhuff committed:
feat(app): Add link support to toasts (#13387)

mjhuff committed:
fix(app): update flex robot software available slideout (#13391)

koji committed:
feat(app): add module calibration section to robotsettings (#13374)

* feat(app): add Module calibration section to RobotSettings

b-cooper committed:
fix(app): adapter locations for intervention modal (#13392)

Update the display location logic for the intervention modals to account for labware in adapters

TamarZanzouri committed:
feature(api): Allow push_out arg in the dispense command (#13365)

vegano1 committed:
feat(api,robot-server): Don't scan for wifi networks on every request and expose option via query parameter (#13405)

ncdiehl11 committed:
feat(app): update instrument and module calibration wizard flows (#13378)

Closes RAUT-586 closes RAUT-647

b-cooper committed:
feat(app): add location select to module calibration wizard (#13408)

Introduce a module location select to the module calibration wizard flow that defaults to the slot
that a module is loaded into within a protocol during setup

---------

Co-authored-by: Shlok Amin <shlokamin@me.com>

Laura-Danielle committed:
feat(shared-data): Support new liquid shape for the update script (#13366)

sfoster1 committed:
fix(api): normal home if fast home stalls (#13417)

If a motor axis says it has a valid encoder position then when we home
we'll try and update the stepper from the encoder, fast move to near
home, and then home. But if the encoder position was wrong - if it
missed some overflows, for instance - then quite possibly that first
fast motion will fail, which right now would fail the home.

Luckily, the fast move is just an optimization. So let's catch a stall
error during the fast move and fall back to a normal home if it happens.
This should be very rare, which is good because "detecting a stall in
the fast move" really means "the fast move crashed into the side of the
robot", but you don't really notice since it's fast.

Closes RQA-1432

sfoster1 committed:
chore(api): remove spammy log (#13420)

This happens a _lot_ now and it is meaningless most of the time so only
log when it's meaningful.

jerader committed:
feat(app): ability to sort protocols via flex or ot-2 (#13419)

closes RQA-1037

b-cooper committed:
fix(app): fix improper capitalization of µL in labware overlays in protocol setup (#13418)

Remove cast to uppercase from labware overlays in labware and liquid setup steps. It reads more
cleanly, and avoids bug where we capitalize "µ" to "M" confusing micro and milli

Closes RAUT-354

SyntaxColoring committed:
perf(robot-server): Handle analysis requests serially to take more advantage of cache (#13397)

SyntaxColoring committed:
fix(robot-server,system-server): Make SQL transactions behave sanely (#13424)

* Use SQLAlchemy's workarounds to fix transactions.

* Deduplicate with system-server.

jerader committed:
feat(app): adapter info in labware & liquid setup  (#13394)

closes RAUT-628

shlokamin committed:
feat(app): update ODD device reset page (#13403)

closes RAUT-639

Co-authored-by: Brian Cooper <brian@opentrons.com>

sanni-t committed:
fix(app): fix non deterministic run command query (#13395)

* use command id for non deterministic runs

koji committed:
feat(app): add tooltip and disabled state to calibrate button in Desktop app (#13414)

* feat(app): add disabled state and tooltip to module calibration button in Desktop app

sfoster1 committed:
fix(api): base plunger speeds on working volume (#13421)


When 62962c599b8b71bccc4c6b18a59a798f8b8989a3 added "liquid classes", we
had to change the way we look up various volumes, and we used the liquid
class max volume as the volume to look up ul/mm transforms for
determining speeds. The thing is, the liquid class max volume didn't
depend on the tip type, but the ul/mm transform did - so if you were
using a 50ul tip on a p1000, you'd try and get the ul/mm for 1000ul on a
50ul tip and get an error.

We can use the working volume instead, which is the maximum volume of
the tip, and fix this up but I think we need a better fix for this that
moves volume constraints into tip definitions.

The flowrate lookups need to actually be looked up _dynamically_ from
the active tip settings rather than cached once when the pipette is created.

The pipette definitions for the OT-2 also got converted to have different
information for tips, but those tips have ul/mm data all the way up to
the maximum volume of the pipette. That lets us stick with the max
volume.

That's also extremely important because the expected flow rates on OT-2
pipettes are those determined by looking up the flowrate from the
pipette nominal volume, so changing to using working volume would alter
the flowrates when using tips of different volumes than default. Thanks,
tests. Thests.

smb2268 committed:
fix(app): add extra protection from run 404 on ODD (#13426)

fix RQA-1424, RQA-1421

vegano1 committed:
refactor(shared-data,hardware-testing): remove calibration adapter and logic for magnetic module. (#13427)

sanni-t committed:
fix(api): 96 channel load name bug (#13415)

vegano1 committed:
fix(api,hardware): add retry mechanism for fw update data packets and can updates. (#13413)

b-cooper committed:
fix(app): find previous labware location from all commands for run preview text (#13370)

In the absence of a way to request timeline-based location information from the protocol engine,
search through all previous commands to reduce labware location by duplicating PE concerns on the
front end.

re RSS-301

fsinapi committed:
fix(usb-bridge): use correct USB gadget configuration strings (#13432)

- Use the serial number on the filesystem, and default to a serial number that follows our serial pattern
- Call the devices a Flex instead of OT3 in the public-facing device information

fsinapi committed:
fix(usb-bridge): worker thread for sending data from tcp to usb (#13411)

* Add retries when sending data from TCP to USB
* Add a worker thread for USB comms
* Log when the queue excessively fills up
* Bound the worker queue

brenthagen committed:
fix(app): use all commands query for ODD intervention modal (#13348)

use all commands query data for ODD intervention modal, as desktop app does

closes RQA-1286

ecormany committed:
refactor(app): update network setup header copy (#13436)

* update network setup header copy

* remove unused string

smb2268 committed:
fix(app): fix run setup pipette bugs (#13440)

fix RQA-1428, RQA-1433

SyntaxColoring committed:
fix(robot-server): Resolve LabwareRole pickle warnings (#13444)

sanni-t committed:
fix(api): readjust hardware controller end state in engine cleanup (#13431)

SyntaxColoring committed:
perf(app,robot-server): Download analyses as raw JSON documents (#13425)

Co-authored-by: Brian Cooper <brian@opentrons.com>

brenthagen committed:
fix(app,app-shell): download robot logs from from app not app-shell (#13388)


makes api requests to download robot logs within the browser instead of using electron-dl package
within app-shell

Closes RQA-1225

jerader committed:
refactor(app): add off deck or on deck to placement (#13437)

closes RAUT-653

fsinapi committed:
fix(hardware): separate very small components of moves into their own steps (#13451)

* De-Diagonalize problematic moves
- Set up a minimum move axis component value of 0.1%
- If a move component is below that number, we split it into its own set of steps
- Added tests for de-diagonilization
* Some function cleanup for readability per Seth's request

sfoster1 committed:
fix(app,api): flex: remove door switch setting (#13454)

On the Flex, the door switch is always enabled. It should not be displayed in the app and default/unset should be read as true on the robot.

Closes RQA-1413

jbleon95 committed:
feat(engine): disallow gripper movement for combination adapter labware (#13428)

Raise error if combination labware/adapter labware is moved with gripper

koji committed:
feat(app): update disabled condition for overflow menu (#13439)

* feat(app): update disabled condition for overflow menu

koji committed:
fix(app): fix white screen in timestamp formatting on run summary (#13438)

* fix(app): fix runsummary whitescreen issue

b-cooper committed:
feat(app): include current labware offsets in touchscreen LPC, implement lpc protocol setup designs (#13445)

Include a link to "view current offsets" in the intro screen of LPC on the ODD. Also fully implement
the designs for the LPC setup step in desktop

Closes RAUT-480

vegano1 committed:
fix(otupdate): use expanduser to get a relative path for authorized_keys. (#13457)

shlokamin committed:
feat(app): wire up module calibration to backend (#13449)

RAUT-655

Co-authored-by: Brian Cooper <brian@opentrons.com>
Co-authored-by: smb2268 <sarah.breen@opentrons.com>

b-cooper committed:
fix(app): fix release notes url in ODD settings (#13456)

Provide full path to opentrons github releases in URL of ODD robot version settings page.

Closes RAUT-658

koji committed:
feat(app): add module calibration to odd (#13423)

* feat(app): add module calibration to odd

koji committed:
fix(app): fix ODD UI bugs (#13455)

* fix(app): fix various odd ui issues

mjhuff committed:
fix(app): fix module cal update banners to show only on flex (#13462)

Closes RAUT-657

shlokamin committed:
feat(app): add app side analytics setting to ODD (#13458)

closes RQA-1303

mjhuff committed:
fix(app): Update ODD maintenance run take over logic (#13461)

* fix(app): update maintenance run takeover logic

Closes RAUT-650

Fixes a bug where starting a maintenance run on the ODD and subsequently on the desktop app for the
same robot would cause the takeover modal not to display on the ODD.

brenthagen committed:
fix(discovery-client): deep compare robot addresses to restart health poller (#13448)

uses lodash isEqual to deep compare previous and current robot addresses to restart health poller.
fixes overeager restart that was causing memory management issues in the electron main process.

koji committed:
fix(app): fix inputField spacing issue on Touchscreen app (#13465)

*fix(app): fix inputField spacing issue on Touchscreen app

ncdiehl11 committed:
fix(app): update send protocol to Flex slideout text (#13464)

* fix(app): update slideout text for sending a protocol to a Flex

- change slideout title and button text

* remove unused translation

* removed duplicate "send" key in protocol_list translator

* pull out Opentrons Flex in translation as constant

* separate translation for overflow sending protocol to flex

* fix test

mjhuff committed:
feat(app, app-shell): desktop: software update flow rework (#13412)

* feat(app): desktop: software update flow rework

Partially closes RAUT-501 

Rework the desktop software update flow. Includes a new download modal, a toast to alert when a
software update is available & successfully installed. Adds new download percentage shell-to-redux pipe. Reworks logic for toggling software update alerts.

SyntaxColoring committed:
fix(robot-server): Pin pickle protocol version to v4 (#13466)

mjhuff committed:
fix(app): fix home gantry on restart setting (#13470)

Closes RQA-1458

Button was not properly toggling although state was updating. Change default display to "on", since
this matches default behavior.

Laura-Danielle committed:
fix(shared-data): Migrate units for mutable configs (#13471)

shlokamin committed:
fix(app): do not check estop status if robot is not a Flex (#13460)

closes RQA-1406

sfoster1 committed:
fix(shared-data): add type ignore in test (#13476)

This is testing a splat that loses types and just doesn't really work
with type annotations sadly. Removing the types fixes the issue.

b-cooper committed:
fix(app): remove hover from empty state in slideouts (#13433)

Remove border color change on hover from the empty state message of the choose robot/protocol
slideouts in the desktop app

Closes RQA-1247

fsinapi committed:
feat(app): acknowledge animation when Flex receives a new protocol (#13474)

* Add schemaV7 definition for StatusBarAnimation control

* When ODD detects a new protocol, run status bar Confirm animation

* Update app/src/App/hooks.ts typo

Co-authored-by: Jethary Rader <66035149+jerader@users.noreply.github.com>

---------

Co-authored-by: Jethary Rader <66035149+jerader@users.noreply.github.com>

jbleon95 committed:
fix(engine): fix labware gripperOffsets not allowing only a default offset (#13472)

jerader committed:
feat(app): create module instruction modal and delete H-S wizard (#13477)

closes RQA-1299

vegano1 committed:
fix(update-server): check that total rootfs size is greater and not equal to the partition size. (#13480)

sfoster1 reviewed (approved):
Thank you! Please remember to normal merge!

koji committed:
fix(app): fix module calibration overflowmenu bug (#13469)

* fix(app): fix module calibration overflowmenu bug and disabled each menu button instead of overflowmenu itself

mjhuff committed:
fix(app): fix overflow menu module calibration buttons (#13482)

Closes RQA-1477

Fixes a bug where pipettes are calibrated but not attached, the calibrate/recalibrate buttons within
the module card overflow menu are enabled.

mjhuff committed:
fix(app): fix error text overlap in slideout menus (#13488)

Closes RQA-1492

Error text was overly nested inside a component, causing text to render within the boundaries of a
sibling component. Fixes the issue across all slideouts.

koji committed:
fix(app): change desktop network icon priority (#13481)

*fix(app): change desktop network icon priority in slideout

koji committed:
fix(app): fix Protocol Run Completed Splash - Title case copy error (#13491)

* fix(app): fix Protocol Run Completed Splash - Title case copy error

mjhuff committed:
fix(app): Fix overeager maintenance run takeover modal (#13483)

* fix(app): fix overeager ODD maintenance run takeover modal

Fixes hanging maintenance run takeover modals after a delete maintenance run request has been sent
to the server.


Co-authored-by: Shlok Amin <shlokamin@me.com>

mjhuff committed:
feat(app)  desktop: robot update flow rework  (#13429)

* feat(app): desktop: robot update flow rework

Closes RAUT-501 and RQA-1430

Rework the desktop robot update flow. Includes new documentation and install modals, progress
percent for download & installation. Refactors existing desktop flow components.
